e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -65,5 +69,112 @@ internal static class TestHelper {
+            Assert.True(!string.IsNullOrWhiteSpace(connectionString), "connection string is null");
+           
+            _registryManager = RegistryManager.CreateFromConnectionString(connectionString);
+            var patch =
+                @"{
+                    tags: {
+                        unmanaged: null
+                    }
+                }";
+
+            UpdateTagAsync(patch).Wait();
+        }
+
+        /// <summary>
+        /// Switch to publisher orchestrated mode
+        /// </summary>
+        /// /// <param name="destinationFilePath">Path of the PublishedNodesFile.json file to be deleted</param>
+        public static void SwitchToOrchestratedMode(string destinationFilePath) {
+            var connectionString = Environment.GetEnvironmentVariable("PCS_IOTHUB_CONNSTRING");
+            Assert.True(!string.IsNullOrWhiteSpace(connectionString), "connection string is null");
+
+            _registryManager = RegistryManager.CreateFromConnectionString(connectionString);
+            var patch =
+               @"{
+                    tags: {
+                        unmanaged: true
+                    }
+                }";
+
+            DeletePublishedNodesFile(destinationFilePath);
+            UpdateTagAsync(patch).Wait();
+        }
+
+        /// <summary>
+        /// Switch to publisher orchestrated mode
+        /// </summary>       
+        /// <param name="deviceId">Password for HTTP basic authentication</param>
+        /// <param name="patch">Name of deployed Industrial IoT</param
+        private static async Task UpdateTagAsync(string patch) {
+            var deviceId = Environment.GetEnvironmentVariable("IOT_EDGE_DEVICE_ID");
+        
+            Assert.True(!string.IsNullOrWhiteSpace(deviceId), "deviceId string is null");
+
+            var twin = await _registryManager.GetTwinAsync(deviceId);
+            await _registryManager.UpdateTwinAsync(twin.DeviceId, patch, twin.ETag);
+        }
+
+        /// <summary>
+        /// transfer the content of published_nodes.json file into the OPC Publisher edge module
+        /// </summary>       
+        /// <param name="sourceFilePath">Source file path</param>
+        /// <param name="destinationFilePath">Destination file path</param
+        public static void LoadPublishedNodesFile(string sourceFilePath, string destinationFilePath) {
+            var username = Environment.GetEnvironmentVariable("PCS_SIMULATION_USER");
+            var password = Environment.GetEnvironmentVariable("PCS_SIMULATION_PASSWORD");
+            var host = Environment.GetEnvironmentVariable("IOT_EDGE_DEVICE_DNS_NAME");
+           
+            Assert.True(!string.IsNullOrWhiteSpace(username), "username string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(password), "password string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(host), "host string is null");
```

Comment:
I would suggest to have a separate `SshHelper` class with its own configuration instead of getting configuration and validating it for each call.

Reply:
added